site/content/docs/tutorials/concierge-and-supervisor-demo.md

```diff
@@ -1,239 +1,654 @@
-    cat <<EOF | kubectl create --context kind-pinniped-concierge -f -
-    apiVersion: authentication.concierge.pinniped.dev/v1alpha1
-    kind: JWTAuthenticator
-    metadata:
-      name: my-jwt-authenticator
-    spec:
-      issuer: $issuer
-      audience: $audience
-      tls:
-        certificateAuthorityData: $(cat /tmp/pinniped-supervisor-ca-bundle-base64-encoded.pem)
-    EOF
-    ```
-
-1. Download the latest version of the Pinniped command-line tool for your platform.
-   On macOS or Linux, you can do this using Homebrew:
-
-   ```sh
-   brew install vmware-tanzu/pinniped/pinniped-cli
-   ```
-
-   On other platforms, see the [command-line installation guide]({{< ref "../howto/install-cli" >}}) for more details.
-
-1. Generate a kubeconfig for the current cluster.
-
-   ```sh
-   pinniped get kubeconfig \
-     --kubeconfig-context kind-pinniped-concierge \
-     > /tmp/pinniped-kubeconfig
-   ```
-
-1. Try using the generated kubeconfig to issue arbitrary `kubectl` commands. The `pinniped` command-line tool
-   opens a browser page that can be used to login to the external OIDC identity provider configured earlier.
-
-   ```sh
-   kubectl --kubeconfig /tmp/pinniped-kubeconfig get pods -n pinniped-concierge
-   ```
-
-   Because this user has no RBAC permissions on this cluster, the previous command results in an
-   error that is similar to
-   `Error from server (Forbidden): pods is forbidden: User "pinny" cannot list resource "pods"
-   in API group "" in the namespace "pinniped"`, where `pinny` is the username that was used to login
-   to the upstream OIDC identity provider. However, this does prove that you are authenticated and
-   acting as the `pinny` user.
-
-1. As the administrator user, create RBAC rules for the test user to give them permissions to perform actions on the cluster.
-   For example, grant the test user permission to view all cluster resources.
-
-   ```sh
-   kubectl --context kind-pinniped-concierge create clusterrolebinding pinny-can-read --clusterrole view --user pinny
-   ```
-
-1. Use the generated kubeconfig to issue arbitrary `kubectl` commands as the `pinny` user.
-
-   ```sh
-   kubectl --kubeconfig /tmp/pinniped-kubeconfig get pods -n pinniped-concierge
-   ```
-
-   The user has permission to list pods, so the command succeeds this time.
-   Pinniped has provided authentication into the cluster for your `kubectl` command. 🎉
-
-1. Carry on issuing as many `kubectl` commands as you'd like as the `pinny` user.
-   Each invocation uses Pinniped for authentication.
-   You may find it convenient to set the `KUBECONFIG` environment variable rather than passing `--kubeconfig` to each invocation.
-
-   ```sh
-   export KUBECONFIG=/tmp/pinniped-kubeconfig
-   kubectl get namespaces
-   kubectl get pods -A
-   ```
+# To delete the service account created above for cert-manager:
+gcloud iam service-accounts delete \
+  "demo-dns-solver@$PROJECT.iam.gserviceaccount.com" \
+  --project "$PROJECT" --quiet
```

Comment:
Do we need a separate command to delete the associated IAM permissions?

Reply:
I tried it and it turns out that if we delete the IAM permissions first and then delete the account, it works better. When I only deleted the account it left the account showing in the GCP UI as deleted. When I delete the permissions first and the account second, then it disappears from the UI.

I'll add that.